server/server_test.go

```diff
@@ -690,10 +690,14 @@ func TestOAuth2ImplicitFlow(t *testing.T) {
 	}
 }
 
-func TestCrossClientScopes(t *testing.T) {
+func setupCrossClientsFixture(t *testing.T, crossClientsTest func(*crossClientsFixture, *crossClientsTestDeferred) (reqDump, respDump []byte)) {
```

Comment:
What does `crossClientsTest` do? What should it perform? Why does it return `reqDump` and `respDump`? Can we just avoid this re-factor in this PR?